msu/systems/persistent_data/persistent_data_system.nut

```diff
@@ -108,4 +109,46 @@
 		result += this.Separator
 		::logInfo(result);
 	}
+
+	function createFile( _fileName, _dataArray )
+	{
+		::MSU.requireInstanceOf(::MSU.Class.ArrayData, _dataArray);
```

Comment:
It's an internal function though because it is called from the system addon. However, if we want this function to be generally callable by other than that one caller from the addon then yes we can change it. Shall we?

Reply:
I don't think we need to change it, but this is just the wrong type here, either remove the require or make it abstract.

Reply:
To clarify, by "change it" I meant make the change you are suggesting i.e. require instance of AbstractData. My point was that because we are using it only internally we just check for the type we send from the addon function. However, if you think this function can be called from places other than the addon, then it makes sense to require instance of AbstractData.